src/exchanges/libs/OrderTaker.sol

```diff
@@ -59,37 +71,38 @@ abstract contract OrderTaker is OrderFiller {
             address[] memory fillApprovalTargets
         ) = __formatFillTakeOrderArgs(
             _targetExchange,
-            _orderAddresses,
-            _orderValues,
-            _orderData,
-            _identifier,
-            _signature
+            _encodedArgs
         );
 
         __fillTakeOrder(
             _targetExchange,
-            _orderAddresses,
-            _orderValues,
-            _orderData,
-            _identifier,
-            _signature,
+            _encodedArgs,
             __encodeOrderFillData(fillAssets, fillExpectedAmounts, fillApprovalTargets)
         );
+
+        IPolicyManager(__getRoutes().policyManager).postValidate(
+            TAKE_ORDER,
+            [
+                riskManagementAddresses[0],
+                riskManagementAddresses[1],
+                riskManagementAddresses[2],
+                riskManagementAddresses[3],
+                _targetExchange
+            ],
+            riskManagementValues,
+            0x0
+        );
     }
 
     // INTERNAL FUNCTIONS
- 
+
     /// @notice Reserved function for executing a take order on an external exchange
     /// @dev When executing your order, use the values in __fillAssets and __fillExpectedAmounts
     /// @dev Include the `validateAndFinalizeFilledOrder` modifier
     /// @param _fillData Encoded data used by the OrderFiller
```

Comment:
Could add natspec here and below for `@param _encodedArgs`, just to explain what these are in the file.